paymentsheet/src/main/java/com/stripe/android/lpmfoundations/paymentmethod/definitions/UsBankAccountDefinition.kt

```diff
@@ -23,15 +24,19 @@ internal object UsBankAccountDefinition : PaymentMethodDefinition {
 
     override fun requiresMandate(metadata: PaymentMethodMetadata): Boolean = true
 
-    override fun supportedPaymentMethod(
-        sharedDataSpec: SharedDataSpec,
-    ): SupportedPaymentMethod {
-        return SupportedPaymentMethod(
-            paymentMethodDefinition = this,
-            sharedDataSpec = sharedDataSpec,
-            displayNameResource = R.string.stripe_paymentsheet_payment_method_us_bank_account,
-            iconResource = R.drawable.stripe_ic_paymentsheet_pm_bank,
-            tintIconOnSelection = true,
-        )
-    }
+    override fun uiDefinitionFactory(): UiDefinitionFactory = UsBankAccountUiDefinitionFactory
+}
+
+private object UsBankAccountUiDefinitionFactory : UiDefinitionFactory.Simple {
+    override fun createSupportedPaymentMethod(): SupportedPaymentMethod = SupportedPaymentMethod(
+        code = UsBankAccountDefinition.type.code,
+        displayNameResource = R.string.stripe_paymentsheet_payment_method_us_bank_account,
+        iconResource = R.drawable.stripe_ic_paymentsheet_pm_bank,
+        tintIconOnSelection = true,
+        lightThemeIconUrl = null,
+        darkThemeIconUrl = null,
+    )
+
+    // US Bank Account uses it's own mechanism, not these form elements.
+    override fun createFormElements(metadata: PaymentMethodMetadata): List<FormElement> = emptyList()
```

Comment:
Thinking long-term, is there an opportunity here to convert to build out a `USBankFormElement` which handles all the US Bank account logic internally? The `USBankAccountForm` composable already wraps quite a bit of that logic. It would be cool if we could pull those definitions out of `PaymentSheet` & `FlowController` entirely.

Reply:
Yup! That's definitely the goal! I don't have plans to tackle that as part of this project, but it's certainly one of the things I wanted to enable! There are further things that could be pulled out of BaseSheetViewModel for us_bank, card (link, cbc, etc), and likely others!

This functionality seems relatively possible now, and could in theory be done in "isolation", if you're interested in doing it, definitely go for it! But if not, it'll be on my list for the long tail of the project (next half/next year).